src/ui/application-window.vala

```diff
@@ -749,6 +756,13 @@ namespace Peek.Ui {
 
     private void handle_saved_file (File file) {
       save_preferred_save_folder (file);
+      string out_file=get_file();
+      string out_file_ext = Utils.get_file_extension_for_format (
+        recorder.config.output_format);
+      ShareDialog.get_file_ext(out_file_ext);
+      ShareDialog.filename(out_file);
```

Comment:
It would probably be easier and more flexible to pass the `File` object to the `ShareDialog` instead of just the URI. Also use `File.get_path()` instead of `file.get_uri()`, see my comment below.